gateway/mw_redis_cache.go

```diff
@@ -222,18 +233,24 @@ func (m *RedisCacheMiddleware) ProcessRequest(w http.ResponseWriter, r *http.Req
 			return nil, http.StatusOK
 		}
 
+		cacheOnlyResponseCodes := m.Spec.CacheOptions.CacheOnlyResponseCodes
+		// override api main CacheOnlyResponseCodes by endpoint specific if provided
+		if cacheMeta != nil &&
+			cacheMeta.CacheOnlyResponseCodes != nil &&
+			len(cacheMeta.CacheOnlyResponseCodes) > 0 {
```

Comment:
only `len(cacheMeta.CacheOnlyResponseCodes) > 0` will do `cacheMeta.CacheOnlyResponseCodes != nil` is redundant.